src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1094,19 +1094,23 @@ private async Task SendInviteAsync(OrganizationUser orgUser, Organization organi
                 throw new BadRequestException("Invalid token.");
             }
 
-            if (string.IsNullOrWhiteSpace(orgUser.Email) ||
-                !orgUser.Email.Equals(user.Email, StringComparison.InvariantCultureIgnoreCase))
-            {
-                throw new BadRequestException("User email does not match invite.");
-            }
-
             var existingOrgUserCount = await _organizationUserRepository.GetCountByOrganizationAsync(
                 orgUser.OrganizationId, user.Email, true);
             if (existingOrgUserCount > 0)
             {
+                if (orgUser.Status == OrganizationUserStatusType.Accepted)
+                {
+                    throw new BadRequestException("You must wait for an administrator to confirm your membership to this organization.");
```

Comment:
Per Trey's comments, `Invitation already accepted. You will receive an email when your Organization membership is confirmed.`

Reply:
The first part of the existing error message (`Unable to accept invitation`) is client-side, and it then appends the rest of the error message supplied by the server. The following error messages (supplied by the server) are appended in this way:
* `User invalid`
* `Invalid token`
* `User email does not match invite`
* `Organization invalid`
* `User not found within organization`
* `You are already part of this organization`

If we want to remove `Unable to accept invitation` for this message (in favour of `Invitation already accepted` which I agree is clearer), we'd have to remove the client-side string and generate the whole error message in the server for all the above errors. Note that this would also lose i18n which is done by client but not server.

Are you OK with this @cscharf?

Reply:
The first part of the existing error message (`Unable to accept invitation`) is client-side, and it then appends the rest of the error message supplied by the server. The following error messages (supplied by the server) are appended in this way:
* `User invalid`
* `Invalid token`
* `User email does not match invite`
* `Organization invalid`
* `User not found within organization`
* `You are already part of this organization`

If we want to remove `Unable to accept invitation` for this message (in favour of `Invitation already accepted` which I agree is clearer), we'd have to remove the client-side string and generate the whole error message in the server for all the above errors. Note that this would also lose i18n which is done by client but not server.

Are you OK with this @cscharf?

Reply:
Nope.... lol, let's just have the final result say,

> Unable to accept invitation. Invitation already accepted. You will receive an email when your Organization membership is confirmed.

So we're not losing all of our translating... 

TBH, on the client we "could" key some of these error messages the server sends and pass them through i18n service, and fallback to what the server sends in English, that way there is at least a chance to make error messages make sense in other languages; but that's not a priority for this particular issue.

Reply:
Roger that:
![Screen Shot 2021-02-17 at 9 17 21 am](https://user-images.githubusercontent.com/31796059/108133971-27b4d300-7101-11eb-95ba-02ac269b5a5e.png)